tools/riemann-aws/bin/riemann-s3-list

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env ruby
+
+require 'riemann/tools'
+
+$0 = __FILE__
+
+class Riemann::Tools::S3Metrics
+  include Riemann::Tools
+  require 'fog'
+  require 'time'
+
+  opt :fog_credentials_file, "Fog credentials file", :type => String
+  opt :fog_credential, "Fog credentials to use", :type => String
+  opt :aws_access, "AWS Access Key", :type => String
+  opt :aws_secret, "AWS Secret Key", :type => String
+  opt :aws_region, "AWS Region", :type => String, :default => "eu-west-1"
+  opt :buckets, "Buckets to pull metrics from, multi=true", :type => String, :multi => true, :required => true
+  opt :max_objects, "Max number of objects to list before stopping to save bandwidth", :default => -1
+
+
+  def tick
+    if options[:fog_credentials_file]
+      Fog.credentials_path = options[:fog_credentials_file]
+      Fog.credential = options[:fog_credential].to_sym
+      connection = Fog::Storage.new
+    else
+      if options[:aws_access] && options[:aws_secret]
+        connection = Fog::Storage.new({
+                                          :provider => "AWS",
+                                          :aws_access_key_id => options[:aws_access],
+                                          :aws_secret_access_key => options[:aws_secret],
+                                          :region => options[:aws_region]
+                                      })
+      else
+        connection = Fog::Storage.new({
+                                          :provider => "AWS",
+                                          :use_iam_profile => true,
+                                          :region => options[:aws_region]
+                                      })
+      end
+    end
+
+    options[:buckets].each do |bucket|
+
+      directory = connection.directories.get(bucket)
+      count = 0
+      directory.files.each do |file|
+        puts file
```

Comment:
Gets the object count of a bucket by actually listing the contents and counting, currently the only way to get these stats with less-than-a-day frequency.